### PR TITLE
Added  IIdentityFactory, ITicketFactory to JwtAuthenticationHandler

### LINKED ISSUE #356: User.Identity.IsAuthenticated is always false when use JWT.Extensions.AspNetCore
Follow the README.md in JWT.Extensions.AspNetCore to configure the JWT Authentication stuff. Then I figured out that `User.Identity.IsAuthenticated` is always false. That is because the default IdentityFactory does not provide the Authentication Theme.
I have to custom  the Identity instance in `options.IdentityFactory` like this:

```c#
 options.IdentityFactory = dictionary =>
  {
       var claims = dictionary.Select(p => new Claim(p.Key, p.Value));
       var identity = new ClaimsIdentity(claims, JwtAuthenticationDefaults.AuthenticationScheme);
       return identity;
   };
```
Does anyone has better idea? Or did I miss configuring something? If my solution is correct. Should the author provide more detail configuration in README.md?

### PR DESCRIPTION
**Pull request by abatishchev**
Resolves #356.

### CODE REVIEW

File: src/JWT.Extensions.AspNetCore/Internal/DefaultIdentityFactory.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ internal static class DefaultIdentityFactory
         public static IIdentity CreateIdentity(IDictionary<string, string> dic)
         {
```
Comment (LeTranAnhVu) [inline src/JWT.Extensions.AspNetCore/Internal/DefaultIdentityFactory.cs:16]: This method is not match with the signature of the IdentityFactory prop in JwtAuthenticationOptions.cs

**Comment by abatishchev** [comment]
@LeTranAnhVu sorry, something went in my way yesterday. Work in progress, will finish later today.

**Comment by abatishchev** [comment]
@LeTranAnhVu can you please try the package from [this build's drop](https://abatishchev.visualstudio.com/OpenSource/_build/results?buildId=884&view=artifacts&pathAsName=false&type=publishedArtifacts)?

**Comment by abatishchev** [comment]
@LeTranAnhVu ping, please.

**Comment by abatishchev** [comment]
@LeTranAnhVu ping, please.

**Comment by abatishchev** [comment]
@LeTranAnhVu ping, please.

**Pull request merged**